homeassistant/components/neato/entity.py

```diff
@@ -17,11 +17,7 @@ class NeatoEntity(Entity):
     def __init__(self, robot: Robot) -> None:
         """Initialize Neato entity."""
         self.robot = robot
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        """Return device info."""
-        return DeviceInfo(
+        self._attr_device_info = DeviceInfo(
```

Comment:
```suggestion
        self._attr_device_info: DeviceInfo = DeviceInfo(
```

This tells mypi that we always expect a `DeviceInfo` and we can avoid the assert statement